ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4301,6 +4301,19 @@ public void recycleDirToCmPath(Path dataPath, boolean isPurge) throws HiveExcept
     }
   }
 
+  private static void deleteAndRename(FileSystem destFs, Path destFile, FileStatus srcStatus, Path destPath)
+          throws IOException {
+    if (destFs.exists(destFile)) {
+      // rename cannot overwrite non empty destination directory, so deleting the destination before renaming.
```

Comment:
This is misleading. destFile need not be directory. Can change such that destFile can be a directory or file.

Reply:
not misleading ..the delete is done for case of directory ...file delete is not required

Reply:
did you mean, it reaches here only for directory?

Reply:
no what i meant is the delete is done (for file and directory) because rename can not handle directory ...